ibis/pandas/execution/window.py

```diff
@@ -130,29 +132,73 @@ def get_aggcontext_window(
     return aggcontext
 
 
+def trim_with_timecontext(data, timecontext):
+    """ Trim data within time range defined by timecontext
+
+        This is a util function used in ``execute_window_op``, where time
+        context might be adjusted for calculation. Data must be trimmed
+        within the original time context before return.
+
+    """
+    df = data.reset_index()
+    name = data.name
+    # Filter the data, here we preserve the time index so that when user is
+    # computing a single column, the computation and the relevant time
+    # indexes are retturned.
+    subset = df.loc[df[TIME_COL].between(*timecontext)]
+    # get index columns for the Series
```

Comment:
can you add a blank line before comments

Reply:
Sure, fixed